backend/src/main/java/com/rota/api/StaffService.java

```diff
@@ -28,10 +33,21 @@
     Instant endTime = end == null ? Instant.MAX : end;
     return engagementRepository.findByStaffId(staffId).stream()
         .filter(engagement ->
-            engagement.getStart().isAfter(startTime)
-          && engagement.getEnd().isBefore(endTime))
+            (engagement.getStart().equals(start)
+                || engagement.getStart().isAfter(startTime))
```

Comment:
This can be refactored to
`!engagement.getStart().isBefore(startTime)`
that way you only call the comparison once. Not sure about the readability though.